### PR TITLE
feat: Confidential compute changes - 2022-10-01-preview

### PR DESCRIPTION
**Pull request by fnuarnav**
Allow user to specify Confidential container sku and specify ConfidentialComputeCcePolicy via annotations

### CODE REVIEW

File: pkg/provider/aci_confidential_test.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
...jogIi8iCiAgICB9LApdCmFsbG93X3Byb3BlcnRpZXNfYWNjZXNzIDo9IHRydWUKYWxsb3dfZHVtcF9zdGFja3MgOj0gdHJ1ZQphbGxvd19ydW50aW1lX2xvZ2dpbmcgOj0gdHJ1ZQphbGxvd19lbnZpcm9ubWVudF92YXJpYWJsZV9kcm9wcGluZyA6PSB0cnVlCmFsbG93X3VuZW5jcnlwdGVkX3NjcmF0Y2ggOj0gdHJ1ZQoKCm1vdW50X2RldmljZSA6PSB7ICJhbGxvd2VkIiA6IHRydWUgfQp1bm1vdW50X2RldmljZSA6PSB7ICJhbGxvd2VkIiA6IHRydWUgfQptb3VudF9vdmVybGF5IDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CnVubW91bnRfb3ZlcmxheSA6PSB7ICJhbGxvd2VkIiA6IHRydWUgfQpjcmVhdGVfY29udGFpbmVyIDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CmV4ZWNfaW5fY29udGFpbmVyIDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CmV4ZWNfZXh0ZXJuYWwgOj0geyAiYWxsb3dlZCIgOiB0cnVlIH0Kc2h1dGRvd25fY29udGFpbmVyIDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CnNpZ25hbF9jb250YWluZXJfcHJvY2VzcyA6PSB7ICJhbGxvd2VkIiA6IHRydWUgfQpwbGFuOV9tb3VudCA6PSB7ICJhbGxvd2VkIiA6IHRydWUgfQpwbGFuOV91bm1vdW50IDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CmdldF9wcm9wZXJ0aWVzIDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CmR1bXBfc3RhY2tzIDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CnJ1bnRpbWVfbG9nZ2luZyA6PSB7ICJhbGxvd2VkIiA6IHRydWUgfQpsb2FkX2ZyYWdtZW50IDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CnNjcmF0Y2hfbW91bnQgOj0geyAiYWxsb3dlZCIgOiB0cnVlIH0Kc2NyYXRjaF91bm1vdW50IDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CnJlYXNvbiA6PSB7ImVycm9ycyI6IGRhdGEuZnJhbWV3b3JrLmVycm9yc30K"
```
Comment (mainaksarcar) [inline pkg/provider/aci_confidential_test.go:20]: The default CCEPolicy may change so hardcoding will not help.
You can fall back to not setting the CCEPolicy and ACI will inject the latest policy and also emit back when you GET

File: e2e/confidential_container_test.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
+package e2e
```
Comment (helayoty) [inline e2e/confidential_container_test.go:1]: Please add the license header.

File: e2e/fixtures/confidential_container_pod.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
...jogIi8iCiAgICB9LApdCmFsbG93X3Byb3BlcnRpZXNfYWNjZXNzIDo9IHRydWUKYWxsb3dfZHVtcF9zdGFja3MgOj0gdHJ1ZQphbGxvd19ydW50aW1lX2xvZ2dpbmcgOj0gdHJ1ZQphbGxvd19lbnZpcm9ubWVudF92YXJpYWJsZV9kcm9wcGluZyA6PSB0cnVlCmFsbG93X3VuZW5jcnlwdGVkX3NjcmF0Y2ggOj0gdHJ1ZQoKCm1vdW50X2RldmljZSA6PSB7ICJhbGxvd2VkIiA6IHRydWUgfQp1bm1vdW50X2RldmljZSA6PSB7ICJhbGxvd2VkIiA6IHRydWUgfQptb3VudF9vdmVybGF5IDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CnVubW91bnRfb3ZlcmxheSA6PSB7ICJhbGxvd2VkIiA6IHRydWUgfQpjcmVhdGVfY29udGFpbmVyIDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CmV4ZWNfaW5fY29udGFpbmVyIDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CmV4ZWNfZXh0ZXJuYWwgOj0geyAiYWxsb3dlZCIgOiB0cnVlIH0Kc2h1dGRvd25fY29udGFpbmVyIDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CnNpZ25hbF9jb250YWluZXJfcHJvY2VzcyA6PSB7ICJhbGxvd2VkIiA6IHRydWUgfQpwbGFuOV9tb3VudCA6PSB7ICJhbGxvd2VkIiA6IHRydWUgfQpwbGFuOV91bm1vdW50IDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CmdldF9wcm9wZXJ0aWVzIDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CmR1bXBfc3RhY2tzIDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CnJ1bnRpbWVfbG9nZ2luZyA6PSB7ICJhbGxvd2VkIiA6IHRydWUgfQpsb2FkX2ZyYWdtZW50IDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CnNjcmF0Y2hfbW91bnQgOj0geyAiYWxsb3dlZCIgOiB0cnVlIH0Kc2NyYXRjaF91bm1vdW50IDo9IHsgImFsbG93ZWQiIDogdHJ1ZSB9CnJlYXNvbiA6PSB7ImVycm9ycyI6IGRhdGEuZnJhbWV3b3JrLmVycm9yc30K"
```
Comment (helayoty) [inline e2e/fixtures/confidential_container_pod.yml:7]: I believe the same comment from @mainaksarcar  will apply here as well.

File: pkg/provider/aci.go [hunk 4]
```diff
@@ -1114,6 +1125,29 @@ func (p *ACIProvider) getContainers(pod *v1.Pod) ([]*azaciv2.Container, error) {
...
+		cg.Properties.SKU = &confidentialSku
+		confidentialComputeProperties := azaciv2.ConfidentialComputeProperties{
+			CcePolicy : &ccePolicy,
+		}
+		cg.Properties.ConfidentialComputeProperties = &confidentialComputeProperties
+	    l.Infof("Setting confidential compute properties with CCE Policy")
+
+	} else if strings.ToLower(containerGroupSku) == "confidential" {
+		cg.Properties.SKU = &confidentialSku
+	    l.Infof("Setting confidential container group SKU")
```
Comment (helayoty) [inline pkg/provider/aci.go:1145]: nit: start log messages with lowercase.

File: pkg/provider/aci.go [hunk 5]
```diff
@@ -1114,6 +1125,29 @@ func (p *ACIProvider) getContainers(pod *v1.Pod) ([]*azaciv2.Container, error) {
...
+
+	l := log.G(ctx).WithField("containerGroup", cg.Name)
+
+	if ccePolicy != "" {
+		cg.Properties.SKU = &confidentialSku
+		confidentialComputeProperties := azaciv2.ConfidentialComputeProperties{
+			CcePolicy : &ccePolicy,
+		}
+		cg.Properties.ConfidentialComputeProperties = &confidentialComputeProperties
+	    l.Infof("Setting confidential compute properties with CCE Policy")
```
Comment (helayoty) [inline pkg/provider/aci.go:1141]: nit: start log messages with lowercase.

**Comment by helayoty** [comment]
lgtm 🚢

**Review by helayoty — APPROVED** [review]

**Pull request merged**